### PR TITLE
Ability to fixate dynamic NumberArray

### PR DESCRIPTION
**Pull request by tinwelint**
when it has done all it's dynamic growing. Fixating a dynamic NumberArray
returns a new instance with better performance, but one which cannot
dynamically grow anymore.

ParallelBatchImporter now fixates the node->relationship cache after
relationship import for faster relationship->relationship stage.

### CODE REVIEW

**Comment by zhenlineo** [comment]
A pretty good change :+1:

**Comment by tinwelint** [comment]
Thanks. A micro-benchmark showed 2-3 orders of magnitude faster `set` on a fixated array.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/DynamicNumberArray.java [hunk 1]
```diff
@@ -28,69 +28,17 @@
...
-    {
-        for ( NumberArray chunk : chunks )
-        {
-            chunk.acceptMemoryStatsVisitor( visitor );
-        }
-    }
-
-    @SuppressWarnings( "unchecked" )
     protected N ensureChunkAt( long index )
     {
```
Comment (zhenlineo) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/DynamicNumberArray.java]: For my understanding, after we call `fixate()`, we are free of this call when we want to visit a chunk at least. So it should be faster.

Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/DynamicNumberArray.java]: Yes and somehow hotspot had trouble inlining things or in some way optimizing things when that was on the path of execution so having a version that doesn't use it made a huge impact.

**Comment by tinwelint** [comment]
@zhenlineo Good to merge?

**Pull request merged**